plugins/woocommerce/client/legacy/js/frontend/single-product.js

```diff
@@ -46,28 +46,49 @@ jQuery( function( $ ) {
 		} )
 		// Star ratings for comments
 		.on( 'init', '#rating', function() {
+			$( 'label[for="rating"]' ).hide();
 			$( '#rating' )
 				.hide()
 				.before(
-					'<p class="stars">\
-						<span>\
-							<a class="star-1" href="#">1</a>\
-							<a class="star-2" href="#">2</a>\
-							<a class="star-3" href="#">3</a>\
-							<a class="star-4" href="#">4</a>\
-							<a class="star-5" href="#">5</a>\
-						</span>\
-					</p>'
+					'<fieldset>\
+						<legend>' + wc_single_product_params.i18n_rating_legend + '&nbsp;<span class="required">*</span></legend>\
+						<p class="stars">\
+							<span>\
+								<a role="radio" tabindex="0" aria-checked="false" class="star-1" href="#">' +
+									wc_single_product_params.i18n_rating_options[0] + 
+								'</a>\
+								<a role="radio" tabindex="-1" aria-checked="false" class="star-2" href="#">' + 
+									wc_single_product_params.i18n_rating_options[1] + 
+								'</a>\
+								<a role="radio" tabindex="-1" aria-checked="false" class="star-3" href="#">' + 
+									wc_single_product_params.i18n_rating_options[2] + 
+								'</a>\
+								<a role="radio" tabindex="-1" aria-checked="false" class="star-4" href="#">' + 
+									wc_single_product_params.i18n_rating_options[3] + 
+								'</a>\
+								<a role="radio" tabindex="-1" aria-checked="false" class="star-5" href="#">' + 
+									wc_single_product_params.i18n_rating_options[4] + 
+								'</a>\
+							</span>\
+						</p>\
+					</fieldset>'
 				);
 		} )
 		.on( 'click', '#respond p.stars a', function() {
 			var $star   	= $( this ),
+				starPos     = $star.closest( 'p.stars' ).find( 'a' ).index( $star ) + 1,
 				$rating 	= $( this ).closest( '#respond' ).find( '#rating' ),
 				$container 	= $( this ).closest( '.stars' );
 
-			$rating.val( $star.text() );
-			$star.siblings( 'a' ).removeClass( 'active' );
-			$star.addClass( 'active' );
+			$rating.val( starPos );
```

Comment:
Set the rating value by getting the star position in the DOM instead of relying on the text value.